contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -889,3 +895,230 @@ get_pltsql_function_signature(PG_FUNCTION_ARGS)
 	ReleaseSysCache(proctup);
 	PG_RETURN_TEXT_P(cstring_to_text(func_signature));
 }
+
+/*
+ * tsql_get_constraint_oid
+ *		Given name and namespace of a constraint, look up the OID.
+ *
+ * Returns InvalidOid if there is no such constraint.
+ */
+Oid
+tsql_get_constraint_oid(char *conname, Oid connamespace)
+{	
+	Relation		tgrel;
+	ScanKeyData 	skey[2];
+	SysScanDesc 	tgscan;
+	HeapTuple		tuple;
+	Oid				result = InvalidOid;
+
+	/* search in pg_constraint by name and namespace */
+	tgrel = table_open(ConstraintRelationId, AccessShareLock);
+	ScanKeyInit(&skey[0],
+				Anum_pg_constraint_conname,
+				BTEqualStrategyNumber, F_NAMEEQ,
+				CStringGetDatum(conname));
+
+	ScanKeyInit(&skey[1],
+				Anum_pg_constraint_connamespace,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(connamespace));
+
+	tgscan = systable_beginscan(tgrel, ConstraintNameNspIndexId,
+								true, NULL, 2, skey);
+
+	/* we are interested in the first row only */
+	if (HeapTupleIsValid(tuple = systable_getnext(tgscan)))
+	{
+		Form_pg_constraint con = (Form_pg_constraint) GETSTRUCT(tuple);
+		if (OidIsValid(con->oid))
+		{
+			result = con->oid;
+		}
+	}
+	systable_endscan(tgscan);
+	table_close(tgrel, AccessShareLock);
+	return result;
+}
+
+/*
+ * tsql_get_trigger_oid
+ *		Given name of a trigger, look up the OID.
+ *
+ * Returns InvalidOid if there is no such trigger.
+ */
+Oid
+tsql_get_trigger_oid(char *tgname)
```

Comment:
Why dont we need to check for schema/namespace for triggers?

Reply:
added check in next commit